### ISSUE TITLE
Ability to open web socket connections from background using net

### ISSUE THREAD

**Issue by tex0l** [electron/electron#8931]
Hi!

I'm currently building an app that needs websockets in the background process. 
There are some packages that do it over Node, but it'd be very convenient if they were handled by `electron.net` so that proxies are handled the same way it currently does for http(s) connections.

Is that possible ?

**Comment by baconbrad** [comment]
I don't quite understand what you are asking.

But if want to have them used in the background then maybe it is as easy as just opening the WS client in the main process. Ref project: https://github.com/116356754/wsobserver

**Comment by tex0l** [comment]
My point is that the `net`component handles automatically proxies in electron, but only for http(s) connections, and I'd like to have something similar for handling proxies for websockets in the main process.

@arantes555 has built https://github.com/arantes555/electron-fetch to have a polyfill of `fetch` in the main process without using an intermediate `BrowserWindow` which is very useful, especially since the chromium implementation of fetch seems to have issues with proxies on Windows (I've reported it on chromium), and we'd like to be able to do the same with websocket connections as well.

In the meantime, we've coded a thing similar to what you proposed to handle websocket connections with an intermediate `BrowserWindow` that communicates through IPC but it's really a hack in my opinion.
